lib/libshout/CMakeLists.txt

```
@@ -37,6 +35,9 @@ add_library(shout_mixxx STATIC
   src/tls.c
 )
 
+target_compile_definitions(shout_mixxx PRIVATE HAVE_CONFIG_H)
+target_compile_options(shout_mixxx PRIVATE -pthread -Wall)
```

Shouldn't we use this? https://cmake.org/cmake/help/latest/module/FindThreads.html

```suggestion
target_link_libraries(shout_mixxx PRIVATE Threads::Threads)
```

Also, why `-Wall`? I don't care for warnings in unrelated non-mixxx code. How about `-w`?

---

I have just restored the upstream build flags. And yes, we "should" care about warnings in everything we build.
But that discussion is out of scope here.

---

-pthread is a compiler option. We can think about optimize this here, but that is out of the scope of this PR.
For now if uses the same flags as upstream and if is Linux only.